src/hooks/useActiveDescendantListFocus/index.js

```diff
@@ -109,7 +118,9 @@ function useActiveDescendantListFocus({
   );
 
   const onBlurCallback = useCallback(() => {
-    setVisualFocusItemIndex(-1);
+    if (visualFocusItemIndex !== -1) {
+      setVisualFocusItemIndex(-1);
+    }
   }, [setVisualFocusItemIndex]);
```

Comment:
Dependencies array should contain `visualFocusItemIndex` instead of `setVisualFocusItemIndex`